pagopa/io-app
# fix: [IOBP-896] Skeletons of loading cards in the Wallet tab

## Short description
This PR addresses an issue where the payment methods in the wallet screen would enter an infinite loading state when the app was brought back to the foreground after being in the background for 1-2 minutes, or when the user session expired. Previously, the only way to resolve this was to restart the app.

With this fix, the payment methods are correctly refreshed and displayed without requiring a restart. Additionally, the custom `useOnWalletUserSessionRefresh` hook has been removed as it is no longer necessary with this solution.

## List of changes proposed in this pull request
- Added a check in the `watchWalletSaga`  to avoid placing placeholders every time the saga is triggered. Previously, placeholders were set every time the watchWalletSaga was invoked (including not only at app startup but also when the user session was renewed), causing payment methods to get stuck in an infinite loading state. Now, a check is performed to see if the payment methods are already populated. If they are, it means the saga has already been triggered previously and the payment methods are loaded, so there is no need to add placeholders or show skeletons again.
- Removed the custom hook useOnWalletUserSessionRefresh, which is no longer needed due to this fix.

## How to test
- With the dev-server started, enable the `fastLogin` feature flag from the `backend.ts` file setting the `min_app_version` to `2.66.0.0` (just to enable it);
- Then open the app IO in dev mode and go inside the Wallet tab;
- The first time you opens this tab, you should be able to see the skeletons;
- Wait 60 seconds in order to trigger the session expiration by switching tab (for instance from Wallet to Messages tab);
- You should be able to see the "renew user session screen";
- When you perform a renew session, switch inside the Wallet section tab again: You shouldn't see the skeletons anymore;

## Preview

|Before|After|
|-|-|
|<video src="https://github.com/user-attachments/assets/2a20d5e3-9d5f-4b69-a30f-8c8e310b0c82" />|<video src="https://github.com/user-attachments/assets/c64f8548-04a0-4c69-abd1-f8cab22ee43d" />
